ui/app/src/components/PreviewDropTargetsPanel/PreviewDropTargetsPanel.tsx

```diff
@@ -127,67 +156,117 @@ export function PreviewDropTargetsPanel() {
+        {allowedContentTypes?.map((contentType: ContentType, i: number) => {
+          return (
+            <ListItemButton
+              key={i}
+              onClick={() => {
+                setListMode(false);
+                handleSelectChange(contentType.id);
+              }}
+            >
+              <ContentTypeItem contentType={contentType} contentTypesBranch={contentTypesBranch} />
+            </ListItemButton>
+          );
+        })}
+      </>
+    ) : (
+      <>
+        <Box className={classes.select} display="flex" alignItems="center">
+          <FormControl>
+            <InputLabel>{formatMessage(translations.selectContentType)}</InputLabel>
+            <Select
+              value={dropTargetsBranch.selectedContentType || ''}
+              label={formatMessage(translations.selectContentType)}
+              onChange={(event) => {
+                event.stopPropagation();
+                handleSelectChange(event.target.value);
+              }}
+            >
+              <ListSubheader>
+                <FormattedMessage defaultMessage="Compatible types" />
+              </ListSubheader>
+              {allowedContentTypes?.map((contentType: ContentType, i: number) => {
+                return (
+                  <MenuItem value={contentType.id} key={i}>
+                    <ContentTypeItem contentType={contentType} contentTypesBranch={contentTypesBranch} />
+                  </MenuItem>
+                );
+              })}
+            </Select>
+          </FormControl>
+          {dropTargetsBranch?.selectedContentType && (
+            <Tooltip title={<FormattedMessage defaultMessage="Cancel selection" />}>
+              <IconButton edge="end" sx={{ marginLeft: '5px' }} onClick={() => resetState()}>
+                <CloseRoundedIcon />
+              </IconButton>
+            </Tooltip>
+          )}
+        </Box>
+        <List>
+          <SuspenseWithEmptyState
+            resource={dropTargetsResource}
+            withEmptyStateProps={{
+              emptyStateProps: {
+                title: dropTargetsBranch.selectedContentType
+                  ? formatMessage(translations.noResults)
+                  : formatMessage(translations.chooseContentType)
+              }
+            }}
+          >
+            <DropTargetsList resource={dropTargetsResource} onSelectedDropZone={onSelectedDropZone} />
+          </SuspenseWithEmptyState>
+        </List>
+      </>
+    )
+  ) : (
+    <EmptyState title="No compatible types were found." sxs={{ title: { textAlign: 'center' } }} />
+  );
+}
+
+interface ContentTypeItemContentProps {
+  contentType: ContentType;
+  contentTypesBranch: EntityState<ContentType>;
+}
+
+function ContentTypeItem(props: ContentTypeItemContentProps) {
+  const { contentType, contentTypesBranch } = props;
+  const theme = useTheme();
+  const { backgroundColor, textColor } = getAvatarWithIconColors(
+    contentTypesBranch.byId[contentType.id].name,
```

Comment:
What do you need the contentTypesBranch for if you already get the contentType object?
```suggestion
interface ContentTypeItemContentProps {
  contentType: ContentType;
}

function ContentTypeItem(props: ContentTypeItemContentProps) {
  const { contentType } = props;
  const theme = useTheme();
  const { backgroundColor, textColor } = getAvatarWithIconColors(
    contentType.name,
```